unit-tests-java8/src/test/java/org/eclipse/collections/test/RichIterableTestCase.java

```diff
@@ -1051,38 +1051,20 @@ default void RichIterable_iterationOrder()
         });
         assertEquals(expectedIterationOrder.size() - 1, toSortedListCount.getCount());
 
-        /*
-        MutableCollection<Integer> toSortedListByIterationOrder = this.newMutableForFilter();
-        this.getInstanceUnderTest().toSortedListBy(toSortedListByIterationOrder::add);
-        assertEquals(expectedIterationOrder.size(), toSortedListByIterationOrder.size());
-        */
-
         Counter toSortedSetCount = new Counter();
         this.getInstanceUnderTest().toSortedSet((o1, o2) -> {
             toSortedSetCount.increment();
             return 0;
         });
         assertEquals(expectedIterationOrder.size(), toSortedSetCount.getCount());
 
-        /*
-        MutableCollection<Integer> toSortedSetByIterationOrder = this.newMutableForFilter();
-        this.getInstanceUnderTest().toSortedSetBy(toSortedSetByIterationOrder::add);
-        assertEquals(expectedIterationOrder.size(), toSortedSetByIterationOrder.size());
-        */
-
         Counter toSortedBagCount = new Counter();
         this.getInstanceUnderTest().toSortedBag((o1, o2) -> {
             toSortedBagCount.increment();
             return 0;
         });
         assertEquals(expectedIterationOrder.size(), toSortedBagCount.getCount());
 
-        /*
-        MutableCollection<Integer> toSortedBagByIterationOrder = this.newMutableForFilter();
-        this.getInstanceUnderTest().toSortedBagBy(toSortedBagByIterationOrder::add);
-        assertEquals(expectedIterationOrder.size(), toSortedBagByIterationOrder.size());
-        */
```

Comment:
Hm, these three are tests for real bugs. I wonder if they are fixed now.

Reply:
I've re-enabled this locally and got the following failures bellow. Should I add those lines back, comment them out and add a todo so that they don't get flagged by that inspection? Should I also open an issue to fix this?


2:30:12:707 [INFO] Results:
12:30:12:707 [INFO] 
12:30:12:707 [ERROR] Failures: 
12:30:12:708 [ERROR]   ListAdapterTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:708 [ERROR]   MultiReaderFastListAsReadUntouchableTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:708 [ERROR]   MultiReaderFastListAsWriteUntouchableTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:708 [ERROR]   RandomAccessListAdapterTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:708 [ERROR]   ImmutableBagTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:709 [ERROR]   ImmutableHashBagTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:709 [ERROR]   ImmutableSortedBagImplTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:709 [ERROR]   HashBagNoIteratorTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:709 [ERROR]   HashBagTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:709 [ERROR]   MultiReaderHashBagTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:710 [ERROR]   SynchronizedMutableBagTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:710 [ERROR]   UnmodifiableBagTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:710 [ERROR]   SynchronizedSortedBagTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:710 [ERROR]   TreeBagNoComparatorTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:710 [ERROR]   TreeBagNoIteratorTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:711 [ERROR]   TreeBagTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:711 [ERROR]   UnmodifiableSortedBagTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:711 [ERROR]   CollectIterableTestNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:711 [ERROR]   CompositeIterableTestNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:712 [ERROR]   DistinctIterableTestNoIteratorTest.RichIterable_iterationOrder expected:<4> but was:<6>
12:30:12:712 [ERROR]   DropIterableNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:712 [ERROR]   DropWhileIterableNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:712 [ERROR]   FlatCollectIterableTestNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:713 [ERROR]   LazyIterableAdapterTestNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:713 [ERROR]   RejectIterableTestNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:713 [ERROR]   SelectInstancesOfIterableTestNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:713 [ERROR]   SelectIterableTestNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:714 [ERROR]   TapIterableTestNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:714 [ERROR]   ImmutableListTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:714 [ERROR]   ArrayListAdapterTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:714 [ERROR]   CompositeFastListTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:715 [ERROR]   FastListNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:715 [ERROR]   FastListTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:715 [ERROR]   MultiReaderFastListTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:716 [ERROR]   SynchronizedMutableListTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:716 [ERROR]   UnmodifiableMutableListTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:717 [ERROR]   ImmutableUnifiedMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:718 [ERROR]   ImmutableTreeMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:718 [ERROR]   ImmutableUnifiedMapWithHashingStrategyTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:719 [ERROR]   ConcurrentHashMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:719 [ERROR]   ConcurrentHashMapUnsafeTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:719 [ERROR]   ConcurrentMutableHashMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:720 [ERROR]   MapAdapterTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:720 [ERROR]   SynchronizedMutableMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:720 [ERROR]   UnifiedMapNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:720 [ERROR]   UnifiedMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:720 [ERROR]   UnmodifiableMutableMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:721 [ERROR]   OrderedMapAdapterTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:721 [ERROR]   UnmodifiableMutableOrderedMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:721 [ERROR]   SortedMapAdapterTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:721 [ERROR]   SynchronizedSortedMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:721 [ERROR]   TreeSortedMapNoIteratorTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:722 [ERROR]   TreeSortedMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:722 [ERROR]   UnmodifiableTreeMapTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:722 [ERROR]   UnifiedMapWithHashingStrategyTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:722 [ERROR]   ImmutableStackTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:723 [ERROR]   ArrayStackTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:723 [ERROR]   SynchronizedMutableStackTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:723 [ERROR]   UnmodifiableMutableStackTest.RichIterable_iterationOrder expected:<10> but was:<18>
12:30:12:723 [INFO] 
12:30:12:723 [ERROR] Tests run: 8047, Failures: 59, Errors: 0, Skipped: 1

Reply:
Sorry for the delay. I think this is good to go as-is. In the meantime other stuff landed so this will need a rebase.